OpenSim/Common/Component.cpp

```diff
@@ -173,16 +173,14 @@ void Component::connect(Component &root)
 
     reset();
 
-    initComponentTreeTraversal(root);
-
     // rebuilding the connectors table, which was emptied by clearStateAllocations
     for (int ix = 0; ix < getProperty_connectors().size(); ++ix){
         AbstractConnector& connector = upd_connectors(ix);
         connector.disconnect();
         try{
-            const Component* connectTo = root.findComponent(connector.get_connected_to_name());
-            connector.connect(*connectTo);
-            //connector.findAndConnect(root);
+            //const Component* connectTo = root.findComponent(connector.get_connected_to_name());
+            //connector.connect(*connectTo);
+            connector.findAndConnect(root);
         }
         catch (...) {
             throw Exception(getConcreteClassName() +
```

Comment:
Since this exception message says "cannot **find** component", maybe it belongs within `findAndConnect()` instead of `connect()`? Otherwise, the language of the exception message could be updated.